src/check/parse/Parser.zig

```diff
@@ -217,32 +218,24 @@ fn parseModuleHeader(self: *Parser) IR.NodeStore.HeaderIdx {
 
     const start = self.pos;
 
-    self.advance();
+    self.advance(); // Advance past KwModule
 
     // Get exposes
-    if (self.peek() != .OpenSquare) {
+    self.expect(.OpenSquare) catch {
         std.debug.panic("TODO: Handle header with no exposes open bracket: {s}", .{@tagName(self.peek())});
-    }
-    self.advance();
-    const scratch_top = self.store.scratchTokenTop();
-    while (self.peek() != .CloseSquare) {
-        if (self.peek() != .LowerIdent and self.peek() != .UpperIdent) {
-            std.debug.panic("TODO: Handler header bad exposes contents: {s}", .{@tagName(self.peek())});
-        }
-
-        self.store.addScratchToken(self.pos);
-        self.advance();
-
-        if (self.peek() != .Comma) {
-            break;
+    };
+    const scratch_top = self.store.scratchExposedItemTop();
+    self.parseCollectionSpan(IR.NodeStore.ExposedItemIdx, .CloseSquare, IR.NodeStore.addScratchExposedItem, Parser.parseExposedItem) catch {
+        std.debug.print("Ended at {s}\n", .{@tagName(self.peek())});
+        while (self.peek() != .CloseSquare and self.peek() != .EndOfFile) {
+            self.advance();
         }
-        self.advance();
-    }
-    if (self.peek() != .CloseSquare) {
-        std.debug.panic("TODO: Handle Bad header no closing exposes bracket: {s}", .{@tagName(self.peek())});
-    }
-    const exposes = self.store.tokenSpanFrom(scratch_top);
-    self.advance();
+        self.expect(.CloseSquare) catch {};
+        self.store.clearScratchExposedItemsFrom(scratch_top);
+        std.debug.print("Expected CloseSquare, got {s}\n", .{@tagName(self.peek())});
```

Comment:
I assume we will modify this in future to be malformed or push a diagnostic?

Reply:
Gone